File: minipack-core/src/test/java/io/github/odenix/minipack/core/example/WriteToStream.java
/*
 * Copyright 2024 the MiniPack contributors
 * SPDX-License-Identifier: Apache-2.0
 */
package io.github.odenix.minipack.core.example;

import org.junit.jupiter.api.Test;
import io.github.odenix.minipack.core.MessageWriter;

import java.io.IOException;
import java.io.OutputStream;

import static org.assertj.core.api.Assertions.assertThat;

class WriteToStream extends Example {
  // -8<- [start:snippet]
  void write(OutputStream stream) throws IOException {
    try (var writer = MessageWriter.of(stream)) {
      writer.write("Hello, MiniPack!");
      writer.write(42);
    }
  }
  // -8<- [end:snippet]

  @Test
  void test() throws IOException {
    write(outStream);
    assertThat(reader.readString()).isEqualTo("Hello, MiniPack!");
    assertThat(reader.readInt()).isEqualTo(42);
  }
}

File: minipack-kotlin/src/test/kotlin/io/github/odenix/minipack/kotlin/example/WriteToStream.kt
/*
 * Copyright 2024 the MiniPack contributors
 * SPDX-License-Identifier: Apache-2.0
 */
package io.github.odenix.minipack.kotlin.example

import io.github.odenix.minipack.kotlin.MessageWriters
import java.io.OutputStream
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class WriteToStream : Example() {
  // -8<- [start:snippet]
  fun write(stream: OutputStream) {
    MessageWriters.of(stream).use { writer ->
      writer.write("Hello, MiniPack!")
      writer.write(42)
    }
  }
  // -8<- [end:snippet]

  @Test
  fun test() {
    write(outStream)
    writer.close()
    assertThat(reader.readString()).isEqualTo("Hello, MiniPack!")
    assertThat(reader.readInt()).isEqualTo(42)
  }
}
